src/scrapers/mizrahi.js

```diff
@@ -0,0 +1,119 @@
+    { selector: '#ctl00_PlaceHolderLogin_ctl00_tbPassword', value: credentials.password },
+  ];
+}
+
+function getPossibleLoginResults() {
+  const urls = {};
+  urls[LOGIN_RESULT.SUCCESS] = [`${AFTER_LOGIN_BASE_URL}`];
+  urls[LOGIN_RESULT.INVALID_PASSWORD] = [`${BASE_URL}/login/loginMTO.aspx`];
+  urls[LOGIN_RESULT.CHANGE_PASSWORD] = [
+    `${AFTER_LOGIN_BASE_URL}/main/uis/ge/changePassword/`,
+  ];
+  return urls;
+}
+
+function CreateDataFromRequest(request, optionsStartDate) {
+  const defaultStartMoment = moment().subtract(1, 'years');
+  const startDate = optionsStartDate || defaultStartMoment.toDate();
+  const startMoment = moment.max(defaultStartMoment, moment(startDate));
+
+  const data = JSON.parse(request.postData());
+
+  data.inToDate = moment().format(DATE_FORMAT);
+  data.inFromDate = startMoment.format(DATE_FORMAT);
+  data.table.maxRow = 9999999999;
+
+  return data;
+}
+
+function createHeadersFromRequest(request) {
+  return {
+    mizrahixsrftoken: request.headers().mizrahixsrftoken,
+    'Content-Type': request.headers()['content-type'],
+  };
+}
+
+function convertTransactions(txns) {
+  return txns.map((row) => {
+    const txnDate = moment(row.MC02PeulaTaaEZ).format(ISO_DATE_FORMAT);
+
+    // TODO: I don't have enough sample transactions to understand the rest of the data.
+    return {
+      type: NORMAL_TXN_TYPE, // can be either 'normal' or 'installments'
+      // identifier: int, // only if exists
+      date: txnDate, // ISO date string
+      processedDate: txnDate, // ISO date string
+      originalAmount: row.MC02SchumEZ,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: row.MC02SchumEZ,
+      description: row.MC02TnuaTeurEZ,
+      // installments: {
+      //   number: int, // the current installment number
+      //   total: int, // the total number of installments
+      // },
+      status: TRANSACTION_STATUS.COMPLETED, // can either be 'completed' or 'pending'
+    };
+  });
+}
+
+class MizrahiScraper extends BaseScraperWithBrowser {
+  getLoginOptions(credentials) {
+    return {
+      loginUrl: `${LOGIN_URL}`,
+      fields: createLoginFields(credentials),
+      submitButtonSelector: '#ctl00_PlaceHolderLogin_ctl00_Enter',
+      postAction: async () => {
+        this.request = await this.page.waitForRequest(TRANSACTIONS_REQUEST_URL);
+      },
+      possibleResults: getPossibleLoginResults(),
+    };
+  }
+
+  async fetchData() {
+    const data = CreateDataFromRequest(this.request, this.options.startDate);
+    const headers = createHeadersFromRequest(this.request);
+
+    const response = await fetchPostWithinPage(this.page,
+      TRANSACTIONS_REQUEST_URL, data, headers);
+
+    if (response.header.success === false) {
+      return {
+        success: false,
+        errorType: 'generic',
+        errorMessage:
+          `Error fetching transaction. Response message: "${response.header.messages[0].text}"`,
```

Comment:
this is only a suggestion feel free to adapt or ignore. Application developers that will use this library most likely show the error as is to the user. The error returned from the server is usually descriptive and localized (Hebrew). When you add a prefix such `Error fetching....` it actually make the UX worse. 

Again only a suggestion from my experience, consider removing your static prefix and return the error message directly.

Reply:
I'm not sure I understand you, but you must know that every time the bank return me an error, it always was `General server error` with no details.
I wanted to add an explanation about that, but I couldn't think of a way to write it.

So the important part is the static message, and the response message is in case it has more information.

Reply:
I see, ok, I thought it is like other scrapers where they provide details like `שם המשתמש או הסיסמא שגויה`